packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -1055,45 +873,156 @@ function* purchaseWithAnything({
       throw new Error('No solana provider / wallet found')
     }
     const sourceWallet = yield* call(provider.connect)
+    const paymentRouterTokenAccount = yield* call(
+      [
+        sdk.services.paymentRouterClient,
+        sdk.services.paymentRouterClient.getOrCreateProgramTokenAccount
+      ],
+      {
+        mint: 'USDC'
+      }
+    )
+    const externalTokenAccountPublicKey = getAssociatedTokenAddressSync(
```

Comment:
You can skip this and the above and simply call `paymentRouterClient.createTransferInstruction()`

Reply:
Not quite. That expects an eth wallet

Reply:
nice. still need to keep these for the jup part, but simpler here